test/reporting/quicksight/lambda/network-interface-check.test.ts

```diff
@@ -0,0 +1,172 @@
+      },
+      {
+        NetworkInterfaceId: 'eni-test11112',
+      },
+      {
+        NetworkInterfaceId: 'eni-test11113',
+      },
+    ],
+  };
+
+  const createEvent = {
+    ...basicCloudFormationEvent,
+    ResourceProperties: {
+      ...basicCloudFormationEvent.ResourceProperties,
+      ...commonProps,
+    },
+
+  };
+
+  const updateEvent = {
+    ...basicCloudFormationUpdateEvent,
+    ResourceProperties: {
+      ...basicCloudFormationEvent.ResourceProperties,
+      ...commonProps,
+    },
+
+  };
+
+  beforeEach(() => {
+    ec2ClientMock.reset();
+  });
+
+  test('NetworkInterface check custom resource test - create', async () => {
+
+    ec2ClientMock.on(DescribeNetworkInterfacesCommand).resolves({
+      NetworkInterfaces: [
+        {
+          NetworkInterfaceId: 'eni-test11111',
+          Status: NetworkInterfaceStatus.in_use,
+        },
+        {
+          NetworkInterfaceId: 'eni-test11112',
+          Status: NetworkInterfaceStatus.in_use,
+        },
+        {
+          NetworkInterfaceId: 'eni-test11113',
+          Status: NetworkInterfaceStatus.in_use,
+        },
+      ],
+    });
+    const resp = await handler(createEvent, context) as CdkCustomResourceResponse;
+    expect(ec2ClientMock).toHaveReceivedCommandTimes(DescribeNetworkInterfacesCommand, 1);
+    expect(resp.Data?.isReady).toBeDefined();
+    expect(resp.Data?.isReady).toEqual(true);
+
+  });
+
+  test('NetworkInterface check custom resource test - update', async () => {
+
+    ec2ClientMock.on(DescribeNetworkInterfacesCommand).resolves({
+      NetworkInterfaces: [
+        {
+          NetworkInterfaceId: 'eni-test11111',
+          Status: NetworkInterfaceStatus.in_use,
+        },
+        {
+          NetworkInterfaceId: 'eni-test11112',
+          Status: NetworkInterfaceStatus.in_use,
+        },
+        {
+          NetworkInterfaceId: 'eni-test11113',
+          Status: NetworkInterfaceStatus.in_use,
+        },
+      ],
+    });
+    const resp = await handler(updateEvent, context) as CdkCustomResourceResponse;
+    expect(ec2ClientMock).toHaveReceivedCommandTimes(DescribeNetworkInterfacesCommand, 1);
+    expect(resp.Data?.isReady).toBeDefined();
+    expect(resp.Data?.isReady).toEqual(true);
+
+  });
+
+  test('NetworkInterface check custom resource test - wait 2 time', async () => {
+
+    ec2ClientMock.on(DescribeNetworkInterfacesCommand).resolvesOnce({
+      NetworkInterfaces: [
+        {
+          NetworkInterfaceId: 'eni-test11111',
+          Status: NetworkInterfaceStatus.available,
+        },
+        {
+          NetworkInterfaceId: 'eni-test11112',
+          Status: NetworkInterfaceStatus.available,
+        },
+        {
+          NetworkInterfaceId: 'eni-test11113',
+          Status: NetworkInterfaceStatus.in_use,
+        },
+      ],
+    }).resolvesOnce({
+      NetworkInterfaces: [
+        {
+          NetworkInterfaceId: 'eni-test11111',
+          Status: NetworkInterfaceStatus.available,
```

Comment:
Why do we only consider the `in_use` not `available` ENI ready for creating data sources in QuickSight?

Reply:
If the network interface is not attached to an instance, the status is available ; 
if a network interface is attached to an instance the status is in-use .

We  consider in-use to make sure that QuickSight VPC connection is 100% percent ready to use.